### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by Rohitprasad83**
- added categories using API on the homepage
- added 404 page
- added private route.
- added redux toolkit for state management
- moved all API calls from services to redux toolkit CreateAsynThunk

### CODE REVIEW

**Review by surbhi84 — APPROVED** [review]
- Code looks clean easy to read.
- Liked that you used redux and thunk.
- Nicely structured folders.
- Looks good to merge.
ps- do add some live links.

File: src/pages/watch-later/WatchLater.jsx [hunk 1]
```diff
@@ -1,25 +1,14 @@
 import { useState, useEffect } from 'react'
-import axios from 'axios'
-import { useAuth } from 'context'
 import { Navbar, Footer, VideoCard } from 'components'
-import { errorToast } from 'components/toast/toasts'
+import { useSelector, useDispatch } from 'react-redux'
+import { getAllWatchLaterVideos } from 'features/watchlater/watchLaterSlice'
```
Comment (arpit01923) [inline src/pages/watch-later/WatchLater.jsx:4]: try barrel exports here in root , i.e. , features

**Review by arpit01923 — APPROVED** [review]
- clean code
- good naming convention 
- try barrel exports

File: src/pages/watch-later/WatchLater.jsx (hunk 1 above)
Comment (Rohitprasad83) [reply src/pages/watch-later/WatchLater.jsx:4]: Sure will do it

**Pull request merged**